src/merkle/path.rs

```diff
@@ -82,3 +82,24 @@ impl IntoIterator for MerklePath {
         self.nodes.into_iter()
     }
 }
+
+// MERKLE PATH CONTAINERS
+// ================================================================================================
+
+/// A container for a [MerklePath] and a [Word] holding the value to which the path opens.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct ValuePath {
+    /// The node value opening for `path`.
+    pub value: Word,
+    /// The path from `value` to `root` (exclusive).
+    pub path: MerklePath,
+}
+
+/// A container for a [MerklePath] and a [Word] holding the value to which the path opens.
```

Comment:
Maybe this?

> A container with a Merkle `path` opening for a newly inserted value, and the new tree `root`

Reply:
Ditto; pushed a small modification to your suggestion